avocado/core/test.py

```diff
@@ -83,7 +83,8 @@ def emit(self, record):
         try:
             msg = self.format(record)
             stream = self.stream
-            stream.write('%s' % msg)
+            stream.write(astring.to_text(msg, self.encoding,
```

Comment:
Yeah, so depending on `msg` it's either "blow up" or "store a slightly different, but lossless version.  OK, let's do 2.